### PR TITLE
Added support for std::bind (redux)

### PR DESCRIPTION
**Pull request by esteve**
This PR adds support for std::bind in function_traits

Extends #175 

Connects to #173

### CODE REVIEW

**Comment by esteve** [comment]
I need to trigger this on Windows to check the std::bind signature, this will fail.

**Comment by esteve** [comment]
http://ci.ros2.org/job/ci_windows_opensplice/786/

**Comment by esteve** [comment]
http://ci.ros2.org/job/ci_windows_opensplice/795/

**Comment by esteve** [comment]
CI jobs:

http://ci.ros2.org/job/ci_linux/731/
http://ci.ros2.org/job/ci_osx/607/
http://ci.ros2.org/job/ci_windows_opensplice/795/

**Comment by esteve** [comment]
I've fixed the `cpplint` and `uncrustify` warnings, and squashed the commits. This is ready for review.

**Comment by dirk-thomas** [comment]
Beside the style comment: lgtm.

File: rclcpp/include/rclcpp/function_traits.hpp [hunk 1]
```diff
@@ -77,6 +78,19 @@ template<typename ReturnTypeT, typename ... Args>
...
 {};
 
+// std::bind
+template<typename ClassT, typename ReturnTypeT, typename ... Args, typename ... FArgs>
+#if defined _LIBCPP_VERSION  // libc++ (Clang)
+struct function_traits<std::__1::__bind<ReturnTypeT (ClassT::*)(Args ...), FArgs ...>>
+#elif defined __GLIBCXX__  // glibc++ (GNU C++)
+struct function_traits<std::_Bind<std::_Mem_fn<ReturnTypeT (ClassT::*)(Args ...)>(FArgs ...)>>
+#elif defined _MSC_VER  // MS Visual Studio
+struct function_traits<std::_Binder<std::_Unforced, ReturnTypeT(__cdecl ClassT::*)(Args ...),
```
Comment (jacquelinekay) [inline rclcpp/include/rclcpp/function_traits.hpp]: http://www.atlnightspots.com/wp-content/uploads/2012/10/mitt-romney-biders-full-of-women-meme-2012-10-16_at_11.01.40_PM.png

Comment (wjwwood) [inline rclcpp/include/rclcpp/function_traits.hpp]: :us:

**Pull request merged**